### PR TITLE
add JSONata node

### PR DESCRIPTION
**Pull request by lublak**

### CODE REVIEW

File: packages/nodes-base/nodes/JSONata/JSONata.node.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
+import { IExecuteFunctions } from 'n8n-core';
+import {
+	IDataObject,
+	INodeExecutionData,
+	INodeType,
+	INodeTypeDescription,
+} from 'n8n-workflow';
+
+import { set } from 'lodash';
+import jsonata from 'jsonata';
```
Comment (Joffcom) [inline packages/nodes-base/nodes/JSONata/JSONata.node.ts:10]: This seems to fail for me with:
```
This module is declared with using 'export =', and can only be used with a default import when using the 'esModuleInterop' flag.
```

Had to change it to:
```
import * as jsonata from 'jsonata';
```

**Comment by Joffcom** [comment]
`"dist/nodes/JSONata/JSONata.node.js",` is missing from package.json > nodes array.

**Comment by lublak** [comment]
@Joffcom thank you :) I will test everything later. I have extracted it from an old self-developed node.
There I have probably not paid proper attention.

**Comment by Joffcom** [comment]
@lublak It is all good, I am sure someone from n8n will give it a review later. Thought I would highlight the 2 changes I had to make to get it running.

**Comment by lublak** [comment]
@Joffcom I am also very grateful because I did not get to test yesterday :)
But I tested it earlier and with your two changes it should run completely.

**Comment by Joffcom** [comment]
Hey @lublak,

Thanks for the contribution, We have now created a Community Node Repository which is our preferred method for new node submissions.

Once a node has been published to the Community Node Repository users will be able to install the node from their n8n interface and will be able to use your node quicker, You will also be able to update the node as needed without needing to wait for the review process.

In the future we may reach out to you about including your node in our main release, But for now I am going to close this PR.

You can find more information in the links below, If you have any questions about this or need any help let us know.

Building Community Nodes: https://docs.n8n.io/integrations/community-nodes/build-community-nodes/
Submitting Community Nodes: https://docs.n8n.io/integrations/creating-nodes/deploy/submit-community-nodes/
Published Nodes: https://www.npmjs.com/search?q=keywords:n8n-community-node-package

**Comment by lublak** [comment]
@Joffcom copy pasta! :P But yes I move it to a community node.

**Comment by cellerich** [comment]
> @Joffcom copy pasta! :P But yes I move it to a community node.

@lublak what happened to the community node with jsonata? did you come around to implement it? was looking for exactly this node, it's so helpful and easy to juggle with json data.

**Comment by lublak** [comment]
> > @Joffcom copy pasta! :P But yes I move it to a community node.
> 
> @lublak what happened to the community node with jsonata? did you come around to implement it? was looking for exactly this node, it's so helpful and easy to juggle with json data.

Hi @cellerich,

I will start working on n8n nodes on 11.01.
There was some issues in my work. We rebuild the complete it department.
I will inform you :) I think I will have it available next week. Sry :/ that I was not available.

**Comment by cellerich** [comment]
> I will start working on n8n nodes on 11.01.

This sounds very exciting. Thanks in advance!

**Comment by lublak** [comment]
@cellerich you can now follow the development here: https://github.com/lublak/n8n-nodes-jsonata :)
I will work on the weekend on this

**Comment by cellerich** [comment]
happy to look into it! Thanks a lot!

**Comment by lublak** [comment]
@cellerich currently waiting for this: https://github.com/jsonata-js/jsonata/issues/613

**Pull request closed without merging**